### PR TITLE
Terminate the stream when the server closes the connection

### LINKED ISSUE #736: Stream stucks if the server closes the channel
Assume that the program forgets to acknowledge the message (or processing of the message takes too long).
After a certain timeout (30 minutes by default) the server closes the channel:

https://www.rabbitmq.com/consumers.html
```
If a consumer does not ack its delivery for more than the timeout value (30 minutes by default), its channel will be closed with a PRECONDITION_FAILED channel exception. The error will be [logged](https://www.rabbitmq.com/logging.html) by the node that the consumer was connected to. All outstanding deliveries on that channel, from all consumers, will be [requeued](https://www.rabbitmq.com/confirms.html#automatic-requeueing).
```

This error will be even propagated to the client but the stream doesn't handle it and will just be idle forever.
I guess that if `AMQPChannel#value.isOpen == false`, the stream must be interrupted. 

Will it be a correct behavior? If yes - I can try to fix it.

### PR DESCRIPTION
**Pull request by simpadjo**
closes #736

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -5,3 +5,5 @@ RabbitMQ:
     - "5672:5672"
   environment:
     - DEBUG=false
+  volumes:
+    - ./rabbit-test-config/:/etc/rabbitmq/
```
Comment (simpadjo) [inline docker-compose.yml:9]: Tweak rabbit configuration so it detects missing ack and terminates the channel faster in tests.
With default settings it takes 30 minutes to reproduce the issue

**Comment by simpadjo** [comment]
@jbwheatley I guess bincompat check failure is related to previous PRs

**Comment by jbwheatley** [comment]
the mima previous version wasn't pointing to the latest

**Comment by jbwheatley** [comment]
could you rebase off main?

**Comment by simpadjo** [comment]
@jbwheatley done

**Review by jbwheatley — APPROVED** [review]

**Comment by simpadjo** [comment]
@jbwheatley thank you for merging! Will you have time to run the release sometime in the nearest future?

**Comment by jbwheatley** [comment]
sorry, looks like releases are blocked by an expired gpg key, and I don't have the relevant permissions on this repo to update it. I've opened an issue and tagged the owners, but in the meantime you may need to publish your own version locally.

**Pull request merged**